py-polars/polars/dataframe/plotting.py

```diff
@@ -29,12 +26,15 @@
 
     Encodings: TypeAlias = Dict[
         str,
-        Union[
-            ChannelX, ChannelY, ChannelColor, ChannelOrder, ChannelSize, ChannelTooltip
-        ],
+        Union[X, Y, Color, Order, Size],
     ]
 
 
+def _add_tooltip(chart: alt.Chart) -> alt.Chart:
+    chart.mark = {"type": chart.mark, "tooltip": True}
+    return chart
```

Comment:
Are marks guaranteed to be strings here? It seems like they are by default in polars, but I'm flagging [just in case you want to do something like this](https://github.com/vega/altair/pull/3394/files#diff-bbe4e187b18d242a366c820d023afd89041759cb96e4ec66c3f34559a72c2f9dR3917-R3922) (which you might already have seen).

Reply:
@joelostblom I'm pretty sure @MarcoGorelli has been peeking at that PR 😉 

Maybe I'm being overly cautious here, but I'm not sure we've tested this [thoroughly enough](https://github.com/vega/altair/pull/3394/files#diff-a0ccf73ccb98eab8693e5522b707705def8cc5bbb5f73f77aab30e2497794b40) for it to be included in `polars` without any tests yet

Reply:
thanks both!

I've reworked this to create `tooltip=` explicitly, reckon this is more robust / robust-enough?

Reply:
Looks good to me too